### ISSUE TITLE
Stack bar chart for timeseries, columnar data  generates overlapped bars

### ISSUE THREAD

**Issue by vishal-kony** [c3js/c3#1329]
The below code generates overlapped bars.

var chart = c3.generate({
    data: {
        x: 'x',
        columns: [
            ['x', '2013-01-01', '2013-01-02', '2013-01-03', '2013-01-04', '2013-01-05', '2013-01-06'],
            ['data1', 30, 200, 100, 400, 150, 250],
            ['data2', 130, 340, 200, 500, 250, 350]
        ],
        colors: {
            data1: '#F00',
            data2: '#000',
        },
        names: {
            data1: "test1",
            data2: "test2",
        },
        types: {
            data1: 'bar',
            data2: 'bar',
        },
        groups: [
        	['data1', 'data2']
        ]
    },
    legend: {
        show: false
    },
    axis: {
        x: {
            type: 'timeseries',
            tick: {
                format: "%e %b %y"
            },
        }
    }
});

please have a look at this one

**Comment by masayuki0812** [comment]
Sorry this is a bug and I think it's has been fixed now.

**Comment by panthony** [comment]
Unfortunately, this case may be resolved but it is still possible to have overlapping bars (stacked or not).

See:

https://jsfiddle.net/mcd6ucge/31/

With the following config:

```javascript
var chart = c3.generate({
    data: {
        x: 'x',
        type: 'bar',
        columns: [
            ['x', '2016-01-05', '2016-01-06', '2016-01-10'],
            ['data1', 50, 56, 75],
            ['data2', 2, 2, 2]
        ],
        groups: [
['data1', 'data2']
]
    },
    axis: {
        x: {
            type: 'timeseries',
            tick: {
                format: '%Y-%m-%d'
            }
        }
    }
});
```

It appears that the bug occurs when the data are not evenly distributed over time. The bars' width are scaled according to the number of values whereas their X position is based on their date.

**Comment by masayuki0812** [comment]
Thanks for checking and opening a new issue. Let's fix this in #1386 and let me close this issue for 0.4.11 release.

**Comment by Ravi-jetani** [comment]
data : { 
        xFormat: '%Y-%m-%d %H:%M:%S',
        columns: []
},
    axis: {
        x: {
            type: 'timeseries',
            tick: {
                format: '%Y-%m-%d %H:%M:%S'
            }
        }
    }

actually we need to specify  xFormat field in data
